src/audio/chain_dma.c

```diff
@@ -333,6 +240,114 @@ static enum task_state chain_task_run(void *data)
 	return SOF_TASK_STATE_RESCHEDULE;
 }
 
+static int chain_task_start(struct comp_dev *dev)
+{
+	struct comp_driver_list *drivers = comp_drivers_get();
+	struct chain_dma_data *cd = comp_get_drvdata(dev);
+	k_spinlock_key_t key;
+	int ret;
+
+	comp_info(dev, "chain_task_start(), host_dma_id = 0x%08x", cd->host_connector_node_id.dw);
+
+	key = k_spin_lock(&drivers->lock);
+	switch (cd->chain_task.state) {
+	case SOF_TASK_STATE_QUEUED:
+		k_spin_unlock(&drivers->lock, key);
+		return 0;
+	case SOF_TASK_STATE_COMPLETED:
+		break;
+	case SOF_TASK_STATE_INIT:
+		break;
+	case SOF_TASK_STATE_FREE:
```

Comment:
Is this addition really necessary, after the task initialization is properly done?

Reply:
After stream resuming it has `SOF_TASK_STATE_FREE`, so we should tret such case the same as `SOF_TASK_STATE_INIT`.